modules/core/shared/src/test/scala/freestyle/tagless/TaglessTests.scala

```diff
@@ -34,6 +34,107 @@ import modules._
 
 class TaglessTests extends WordSpec with Matchers {
 
+  "the @tagless macro annotation should be accepted if it is applied to" when {
+
+    "a trait with at least one request" in {
+      "@tagless trait X { def bar(x:Int): FS[Int] }" should compile
+    }
+
+    "a trait with an F[_] bound type param" in {
+      "@tagless @debug trait FBound[F[_]] { def bar(x:Int): FS[Int] }" should compile
```

Comment:
```diff
- @tagless @debug trait FBound[F[_]] { def bar(x:Int): FS[Int] }
+ @tagless @debug trait FBound[F[_]] { def bar(x:Int): F[Int] }
```

Shouldn't be in that way?

Reply:
No. Request methods in a `@free` or `@tagless` are identified by the use of the  `FS` type constructor, to distinguish them from the non-request methods.